feat: WIP multiple DBs IT job

stoerti opened the pull request:
## Description

This PR adds integration test steps against additional RDBMS databases: MySQL, MariaDB, Oracle, MSSQL
- the tests for these vendors are executed every day in the scheduled morning CI build 

Additional changes needed for this:
- refactored the RDBMS exclusion pattern for integration tests
- added random DDL prefix for rdbms ITs (for inmemory H2 was that was not needed so far)
- fixed various smaller bugs in the SQL statements for the additional vendors
  - datatype mappings regarding datetype / timestamp
  - SQL syntax errors
  - added missing DDL prefixes
- adjusted integration tests to match some very special database behavior:
  - MySQL and MariaDB have different sorting collations for UTF8 (they sort case-insensitive) => discussion ongoing of that's a problem
  - Oracle treats and `UPDATE table SET col = ''` as `UPDATE table SET col = NULL` (that can't be changed!)

## Related issues

closes #36365

Follow-Ups:
- https://github.com/camunda/camunda/issues/39888

stoerti committed:
feat: Refactor multi DB integration tests, add option to configure different RDBMS #36365

Introduce changed regex for excluding rdbms in ITests

stoerti committed:
feat: Align versions and configuration if RDBMS vendor images with test configuration #36365

stoerti committed:
feat: Fix various SQL problems with different database vendors #36365

stoerti committed:
feat: Adjust tests to comply with special behavior of different rdbms database vendors #36365

stoerti committed:
feat: Add integration test runs for all other vendors to CI pipeline #36365

stefanzilske reviewed (approved):
Before merge, complete the TODOs in `ci.yaml` and make sure to reset the default in `MultiDbTest` back to LOCAL - apart from that LGTM 👍

stoerti committed:
feat: copilot review

stoerti committed:
feat: activate multi-vendor RDBMS ITs only on scheduled pipelines

kalmarz reviewed (changes requested):
Left comment for a possible solution to the copy-paste problems

stoerti committed:
feat: use reusable job to trigger secondary database IT

stoerti commented:
Thanks for the feedback and suggestions @kalmarz.

Unfortunately it seems, that reusable jobs and `services` are not working together. The yaml schema simply does not support to define services in a job next to a `uses` clause. Also the internet (and copilot) said, that this is not possible.

Still I found another solution, which seems to work well. Now I don't start the needed service as a github action service but as a regular docker-compose action from already existing docker-compose files in the repository. With this I still can control which database to start and reduce the amount of duplicated code.

Because of how Github actions work, I needed to create a separate PR (https://github.com/camunda/camunda/pull/41342) for adding the reusable action, so I can use this action in this PR without the specific branch reference (after merging into main). So please have a look there too.

stoerti committed:
fix: fixed reading authorizations

Remove resource_name as part of the matcher since resource_id is already present and resource_name is nullable (NULL != NULL in Oracle)

stoerti committed:
fix: use former ES/OS properties from pipeline on docker-compose.yml

stoerti committed:
fix: add missing MyBatis datatype mapping

stoerti committed:
fix: declare outputs for database-it reusable

stoerti committed:
ci: test against es/os/h2 on every java code change, other rdbms only once each day

kalmarz commented:
The CI looks good now, thank you!

cmur2 commented:
@stoerti Do you plan to make the newly added GHA jobs in `ci.yml` run also on code changes in the future? Currently, I see 5 of them only run once each morning.

I'm asking because having GHA jobs in the `ci.yml` that are not triggered on code changes (but only on schedule) is an anti-pattern against the [Unified CI](https://github.com/camunda/camunda/wiki/CI-&-Automation#unified-ci) goal. The Unified CI is the quality gate for Pull Requests and the merge queue.

The new jobs (running only on schedule) should reside in a separate GHA workflow file to keep the design leaner.
